src/main/webapp/app/shared/user-settings/learner-profile/course-learner-profile.component.ts

```diff
@@ -0,0 +1,143 @@
+        }
+    }
+
+    onCourseLearnerProfileUpdateError(error: HttpErrorResponse, stateSignal: WritableSignal<EditStateTransition>) {
+        stateSignal.set(EditStateTransition.Abort);
+
+        const errorMessage = error.error ? error.error.title : error.headers?.get('x-artemisapp-alert');
+        if (errorMessage) {
+            this.alertService.addAlert({
+                type: AlertType.DANGER,
+                message: errorMessage,
+                disableTranslation: true,
+            });
+        }
+    }
+
+    updateAimForGradeOrBonus(value: number) {
+        if (!this.courseLearnerProfiles || this.aimForGradeOrBonusState() != EditStateTransition.TrySave) {
+            return;
+        }
+        const clp = this.courseLearnerProfiles[this.activeCourse];
+        clp.aimForGradeOrBonus = value;
+        this.learnerProfileAPIService.putUpdatedCourseLearnerProfile(clp).then(
+            (courseLearnerProfile) => {
+                this.courseLearnerProfiles[this.activeCourse] = courseLearnerProfile;
+                this.aimForGradeOrBonusState.set(EditStateTransition.Saved);
+            },
+            (res: HttpErrorResponse) => this.onCourseLearnerProfileUpdateError(res, this.aimForGradeOrBonusState),
+        );
+    }
+
+    updateTimeInvestment(value: number) {
+        if (!this.courseLearnerProfiles || this.timeInvestmentState() != EditStateTransition.TrySave) {
+            return;
+        }
+        const clp = this.courseLearnerProfiles[this.activeCourse];
+        clp.timeInvestment = value;
+        this.learnerProfileAPIService.putUpdatedCourseLearnerProfile(clp).then(
+            (courseLearnerProfile) => {
+                this.courseLearnerProfiles[this.activeCourse] = courseLearnerProfile;
+                this.timeInvestmentState.set(EditStateTransition.Saved);
+            },
+            (res: HttpErrorResponse) => this.onCourseLearnerProfileUpdateError(res, this.timeInvestmentState),
+        );
+    }
+
+    updateRepetitionIntensity(value: number) {
+        if (!this.courseLearnerProfiles || this.repetitionIntensityState() != EditStateTransition.TrySave) {
+            return;
+        }
+        const clp = this.courseLearnerProfiles[this.activeCourse];
+        clp.repetitionIntensity = value;
+        this.learnerProfileAPIService.putUpdatedCourseLearnerProfile(clp).then(
+            (courseLearnerProfile) => {
+                this.courseLearnerProfiles[this.activeCourse] = courseLearnerProfile;
+                this.repetitionIntensityState.set(EditStateTransition.Saved);
+            },
+            (res: HttpErrorResponse) => this.onCourseLearnerProfileUpdateError(res, this.repetitionIntensityState),
+        );
+    }
+
+    async loadProfiles() {
+        this.courseLearnerProfiles = await this.learnerProfileAPIService.getCourseLearnerProfilesForCurrentUser();
+    }
+
+    loadCourses() {
+        this.courseManagementService.findAllForDropdown().subscribe({
```

Comment:
it's unclear why you cann `findAllForDropdown` here. In general I would prefer more inline code comments to better understand what's going on.